libs/platform/experience/carousel-navigation/carousel-navigation.component.ts

```diff
@@ -0,0 +1,410 @@
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { throttle } from '@spryker-oryx/utilities';
+import { LitElement, PropertyValueMap, TemplateResult, html } from 'lit';
+import { property, queryAll, state } from 'lit/decorators.js';
+import {
+  ArrowNavigationBehavior,
+  CarouselIndicatorAlignment,
+  CarouselIndicatorPosition,
+  CarouselLayoutProperties,
+  CarouselScrollBehavior,
+} from '../src/services';
+import { carouselNavigationStyles } from './carousel-navigation.styles';
+
+export class CarouselNavigationComponent
+  extends LitElement
+  implements CarouselLayoutProperties
+{
+  static styles = carouselNavigationStyles;
+
+  @property({ type: Boolean }) showArrows?: boolean;
+  @property({ type: Boolean }) showIndicators?: boolean;
+  @property() arrowNavigationBehavior?: ArrowNavigationBehavior;
+  @property({ reflect: true }) indicatorsPosition?: CarouselIndicatorPosition;
+  @property({ reflect: true }) indicatorsAlignment?: CarouselIndicatorAlignment;
+  @property() scrollBehavior?: CarouselScrollBehavior;
+
+  protected resizeObserver?: ResizeObserver;
+  protected intersectionObserver?: IntersectionObserver;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  protected throttledScrollListener: () => void = () => {};
+
+  @state() protected items: HTMLElement[] = [];
+  @state() protected slides: { index: number }[] = [];
+
+  @queryAll('input') protected indicatorElements?: NodeListOf<HTMLInputElement>;
+
+  /**
+   * @override Initialize the intersection observer. The intersection observer is
+   * used to detect when the carousel becomes visible and sets the resize observer
+   * and listens to scroll events to always keep the state up to date.
+   */
+  connectedCallback(): void {
+    this.initializeIntersectionObserver();
+    super.connectedCallback();
+  }
+
+  disconnectedCallback(): void {
+    this.resizeObserver?.disconnect();
+    this.intersectionObserver?.disconnect();
+    if (this.hostElement) {
+      this.hostElement.removeEventListener(
+        'scroll',
+        this.throttledScrollListener
+      );
+      this.hostElement.removeEventListener('scrollend', this.updateState);
+    }
+    super.disconnectedCallback();
+  }
```

Comment:
seems to be that subscription logic can be moved to the controller, to decrease complexity

Reply:
I've actually tried that, but it became a bit messy at the end. I/we could do another try.

Reply:
won't do for now.